### PR TITLE
Feature pan validation

### PR DESCRIPTION
**Pull request by kunal-adobe**
Test URLs:

- https://feature-pan-validation--hdfc-bank--aemsites.hlx.page/content/forms/af/hdfc

### CODE REVIEW

**Review by BazilMuralidharan — APPROVED** [review]

File: creditcards/corporate-creditcardFunctions.js [hunk 1]
```diff
@@ -572,7 +588,208 @@ const RESENDOTP = {
...
+      executeInterfaceApi();
+    } else {
+      APS_PAN_CHK_FLAG = 'Y';
+      executeInterfaceApi();
+    }
+  };
+
+  // Main logic to check user proceed status
+  if (IS_ETB_USER) {
+    if (CUSTOMER_INPUT.mobileNumber) {
```
Comment (srivastava21ashish) [inline creditcards/corporate-creditcardFunctions.js:661]: Unecessary cyclometic complexity added. This is not a condition but a switch

File: creditcards/corporate-creditcardFunctions.js [hunk 2]
```diff
@@ -572,7 +588,208 @@ const RESENDOTP = {
...
+              enableFormField('field-pannumberpersonaldetails');
+              console.log('retry');
+            }
+          }
+        } else {
+          executeCheck();
+        }
+      }
+    }
+  } else if (CUSTOMER_INPUT.mobileNumber) {
```
Comment (srivastava21ashish) [inline creditcards/corporate-creditcardFunctions.js:700]: what are we doing here?

File: creditcards/corporate-creditcardFunctions.js [hunk 3]
```diff
@@ -572,7 +588,208 @@ const RESENDOTP = {
...
+          panInfo: {
+            panNumber: personalDetails.panNumberPersonalDetails.$value,
+            panType: 'P',
+            dob: convertDateToDdMmYyyy(new Date(personalDetails.dobPersonalDetails.$value)),
+            name: personalDetails.firstName.$value ? personalDetails.firstName.$value.split(' ')[0] : '',
+          },
+        };
+        return reqObj;
+      } catch (errObj) {
+        const errorCode = 'INVALID_INPUT_PAN_REQUEST';
```
Comment (srivastava21ashish) [inline creditcards/corporate-creditcardFunctions.js:738]: where are we using this constant?

File: creditcards/corporate-creditcardFunctions.js [hunk 4]
```diff
@@ -572,7 +588,208 @@ const RESENDOTP = {
...
+          mobileNumber: globals.form.loginPanel.mobilePanel.registeredMobileNumber.$value,
+          panInfo: {
+            panNumber: personalDetails.panNumberPersonalDetails.$value,
+            panType: 'P',
+            dob: convertDateToDdMmYyyy(new Date(personalDetails.dobPersonalDetails.$value)),
+            name: personalDetails.firstName.$value ? personalDetails.firstName.$value.split(' ')[0] : '',
+          },
+        };
+        return reqObj;
+      } catch (errObj) {
```
Comment (srivastava21ashish) [inline creditcards/corporate-creditcardFunctions.js:737]: arrow function please!

File: creditcards/corporate-creditcardFunctions.js [hunk 5]
```diff
@@ -572,7 +588,208 @@ const RESENDOTP = {
...
+        }
+      } catch (errObj) {
+        console.log(errObj);
+      }
+    },
+    /**
+     * Handles failed PAN validation response.
+     * @param {Object} errorObj - The error object containing details of the failure.
+     */
+    handlePanFailure(errorObj) {
```
Comment (srivastava21ashish) [inline creditcards/corporate-creditcardFunctions.js:780]: Proper nomencalture please! This error object name doesn't tell anything about what this error is. 
Likewise on overall file changes

**Review by srivastava21ashish — CHANGES_REQUESTED** [review]

**Review by srivastava21ashish — APPROVED** [review]

**Pull request merged**